go/samples/coffee-shop/main.go

```diff
@@ -151,6 +165,51 @@ func main() {
 		return text, nil
 	})
 
+	r := &jsonschema.Reflector{
```

Comment:
We should design an API to do this for people. Like maybe another field in dotprompt.Config that you can set to a value to reflect, instead of setting InputSchema. WDYT?

Reply:
This is a good idea. This PR is getting big so I added it to the 3 other TODOs to do in a follow up.